public/app/plugins/panel/xychart/utils.ts

```diff
@@ -62,8 +62,8 @@ export function prepSeries(
 
     let xMatcher = getFieldMatcher(
       seriesCfg.x?.matcher ?? {
-        id: FieldMatcherID.byTypes,
-        options: new Set(['number', 'time']),
+        id: FieldMatcherID.byType,
+        options: 'number',
```

Comment:
fixes a regression with automatically picking up the first/second number fields for x and y axes. this will no longer try to auto-choose time field for an axis, but still allows it to be chosen explicitly in panel edit selector which would set the explicit matcher.

regressed in https://github.com/grafana/grafana/pull/106459